src/app/workflow/view/view.component.html

```diff
@@ -17,6 +17,16 @@
 <button type="button" mat-button color="accent" class="btn-block" (click)="showVersionModal()">
   {{ isPublic || !canWrite || (entryType$ | async) === EntryType.Service ? 'View' : 'Edit' }}
 </button>
+<button
+  type="button"
+  mat-button
+  color="accent"
+  class="btn-block dockstore-snapshot"
+  *ngIf="!version.frozen && version.referenceType !== 'BRANCH'"
+  (click)="snapshotVersion()"
```

Comment:
disable button upon "isRefreshing" to prevent someone with a really slow internet connection from mashing the button

Reply:
Going to do this as part of https://github.com/dockstore/dockstore/issues/2646 will need a little more explanation of the right way to do it.